packages/next/shared/lib/dynamic.tsx

```diff
@@ -41,22 +44,32 @@ export type LoadableFn<P = {}> = (
 export type LoadableComponent<P = {}> = React.ComponentType<P>
 
 export function noSSR<P = {}>(
-  LoadableInitializer: LoadableFn<P>,
+  _LoadableInitializer: LoadableFn<P>,
   loadableOptions: DynamicOptions<P>
 ): React.ComponentType<P> {
   // Removing webpack and modules means react-loadable won't try preloading
   delete loadableOptions.webpack
   delete loadableOptions.modules
 
-  // This check is necessary to prevent react-loadable from initializing on the server
-  if (!isServerSide) {
-    return LoadableInitializer(loadableOptions)
-  }
+  const NoSSRComponent = React.lazy(
+    (isServerSide
+      ? async () => ({ default: () => null })
```

Comment:
For noSSR, the equivalent is not to return null but to throw an error. That will put the Suspense boundary in a client-rendered state on the server.

So this should be something like `() => { throw new Error('This lazy component is intentionally client rendered only.') }`.